beacon-chain/blockchain/process_block.go

```diff
@@ -413,3 +417,29 @@ func (s *Service) savePostStateInfo(ctx context.Context, r [32]byte, b block.Sig
 	}
 	return nil
 }
+
+// This removes the attestations from the mem pool. It will only remove the attestations if input root `r` is canonical,
+// meaning the block `b` is part of the canonical chain.
+func (s *Service) removeBlockAttestationsInPool(ctx context.Context, r [32]byte, b block.SignedBeaconBlock) error {
```

Comment:
Opted in to save one HTR by having the caller passing in precomputed hash tree root `r`

Another option is to `b.Block().HashTreeRoot()` within the function but not sure about the performance implication during batch block processing so I opted out